trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/TrinoS3ProxyClient.java

```diff
@@ -106,6 +112,25 @@ public void proxyRequest(SigningMetadata signingMetadata, ContainerRequest reque
                 .session()
                 .ifPresent(sessionToken -> remoteRequestHeaders.add("x-amz-security-token", sessionToken));
 
+        if (request.hasEntity()) {
+            InputStream entityStream;
+            if ("aws-chunked".equals(request.getRequestHeaders().getFirst("content-encoding"))) {
+                // AWS's custom chunked encoding doesn't get handled by Jersey. Do it manually.
+                try {
+                    entityStream = new ChunkedInputStream(request.getEntityStream());
+                }
+                catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            }
+            else {
+                entityStream = request.getEntityStream();
+            }
+
+            remoteRequestBuilder.setBodyGenerator(new StreamingBodyGenerator(entityStream));
+            remoteRequestHeaders.putSingle("x-amz-content-sha256", "UNSIGNED-PAYLOAD");
```

Comment:
I believe that `x-amz-content-sha256` set to `UNSIGNED-PAYLOAD` is only applicable for single-chunk uploads ([docs](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html)) . For multi-chunk uploads it seems to work differently as outlined in [here](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html#example-signature-calculations-streaming).

Am I misunderstanding the signature logic?

Reply:
I think that doc is misleading. FWIW - Minio fails if I don't use `UNSIGNED-PAYLOAD `, I don't think the number of chunks matters. The fact is we aren't signing anything.

Reply:
Yeah - I think there's a problem here. I may need to add a chunked wrapper. Stay tuned.

Reply:
I did a ton of searching today and I believe that a single-chunk upload will work for us. All the docs I read imply that you can choose to upload a single chunk (up to normal single PUT limits). The other part of this is that I have yet to get proper aws-chunked to work - it's a bear. So, I say we stick with a single unsigned chunk for now until we find out we need to support something else. wdyt?